### PR TITLE
[urduhack] add urduhack

### PR DESCRIPTION
**Pull request by muhammadfahid51**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by muhammadfahid51** [comment]
@synapticarbors I have done what you said. Still it is giving me the same error. I think my package is not installing the requirements.

**Comment by muhammadfahid51** [comment]
@synapticarbors it gets stuck at different point in windows, mac and linux.

File: recipes/urduhack/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - bspatch4 = bsdiff4.cli:main_bspatch4
+
+requirements:
+  host:
+    - pip
+    - python
+    - regex
+  run:
+    - python
+
```
Comment (kmuehlbauer) [inline recipes/urduhack/meta.yaml:27]: @muhammadfahid51 If your package uses `regex` at runtime, you need to list it in the `run`-section.

File: recipes/example/meta.yaml [hunk 2]
```diff
@@ -63,7 +63,7 @@ about:
   # It is strongly encouraged to include a license file in the package,
   # (even if the license doesn't require it) using the license_file entry.
   # See https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#license-file
-  license_file: LICENSE.txt
```
Comment (marcelotrevisani) [inline recipes/example/meta.yaml:66]: please revert this modification

**Comment by synapticarbors** [comment]
It looks like you still need `tf2crf` (see: `https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=200464&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=1102), since it is imported when you import the module.

Also, for the entry points, it looks like you just copied the examples directly both in the definition of the entry points and in the tests:

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-commands

If should be something like:

```
entry_points:
    - urduhack = urduhack.command.cli:cli 
```

and 

```
test:
  commands:
    - urduhack -h
```

if the `urduhack` cli takes a `-h` flag to print its usage. If not `--help` or whatever allows to to run and have a zero exit status.

File: recipes/urduhack/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: 9170ec2e20a6fd2f0620a6d24d61c4233b2b4afe5c1b7e4d4326ef8f04cc6cb4
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+  entry_points:
+    - bsdiff4 = bsdiff4.cli:main_bsdiff4
+    - bspatch4 = bsdiff4.cli:main_bspatch4
+  osx_is_app: True
```
Comment (synapticarbors) [inline recipes/urduhack/meta.yaml:19]: I'm not sure that you need to set `osx_is_app`

File: recipes/urduhack/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 9170ec2e20a6fd2f0620a6d24d61c4233b2b4afe5c1b7e4d4326ef8f04cc6cb4
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+  entry_points:
+    - urduhack.command.cli:cli
```
Comment (synapticarbors) [inline recipes/urduhack/meta.yaml:17]: `- urduhack = urduhack.command.cli:cli `

as noted in my comment

**Comment by synapticarbors** [comment]
@muhammadfahid51 -- you don't need to remove the `noarch`. The two issues that I can see are:

- fixing the entry_point` as I described
- `tf2crf` is not available so it needs to be created in another pull request before this will build

**Comment by muhammadfahid51** [comment]
> @muhammadfahid51 -- you don't need to remove the `noarch`. The two issues that I can see are:
> 
>     * fixing the entry_point` as I described
> 
>     * `tf2crf` is not available so it needs to be created in another pull request before this will build

I have added back noarch and also tf2crf to run

**Comment by muhammadfahid51** [comment]
> @muhammadfahid51 -- you don't need to remove the `noarch`. The two issues that I can see are:
> 
>     * fixing the entry_point` as I described
> 
>     * `tf2crf` is not available so it needs to be created in another pull request before this will build

I think tf2crf is not available on conda. 

```
conda.exceptions.ResolvePackageNotFound: 
  - tf2crf


During handling the above exception

conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {'tf2crf'}
```

Is there a solution for this ?

**Comment by synapticarbors** [comment]
As I mentioned in https://github.com/conda-forge/staged-recipes/pull/12442#issuecomment-679078538, `tf2crf` is not available on conda-forge or in the `defaults` channel, so another pull request will have to be submitted to build it before this package will work. 

The problem is that `tf2crf` also has a dependencies, `tensorflow_addons`, that also does not appear to be available. So building up the dependency chain is going to take some effort.

**Pull request closed without merging**